src/formElementPlugins/dropdownChoice/src/ApplicationView.tsx

```diff
@@ -30,43 +28,46 @@ const ApplicationView: React.FC<ApplicationViewProps> = ({
     options,
     optionsDisplayProperty,
     hasOther,
-    default: defaultOption,
+    default: defaultValue,
+    replaceResponseOnDefaultChange,
   } = parameters as {
     label?: string
     description?: string
     placeholder?: string
     search?: boolean
     hasOther?: boolean
-    default: any
+    default?: string | number
+    replaceResponseOnDefaultChange?: boolean
   } & (ObjectOptions | StringOptions)
 
-  const [selectedIndex, setSelectedIndex] = useState<number>()
+  const [selectedIndex, setSelectedIndex] = useState<number | undefined>(
+    currentResponse?.optionIndex ?? undefined
+  )
   const [addedOption, setAddedOption] = useState<string | null>(null)
 
   const { isEditable } = element
 
   useEffect(() => {
-    // This ensures that, if a default is specified, it gets saved on first load
-    if (!currentResponse?.text && defaultOption !== undefined) {
-      const optionIndex = getDefaultIndex(defaultOption, options)
-      onSave({
-        text:
-          optionsDisplayProperty &&
-          options[optionIndex] !== undefined &&
-          typeof options[optionIndex] === 'object'
-            ? (options[optionIndex] as ObjectOption)[optionsDisplayProperty]
-            : options[optionIndex],
-        selection: options[optionIndex],
-        optionIndex,
-        isCustomOption: false,
-      })
-      setSelectedIndex(optionIndex)
-    }
-    if (currentResponse?.text) {
-      const { optionIndex } = currentResponse
-      setSelectedIndex(optionIndex)
+    // This deals with the case when a default response has been externally set
+    // (e.g. by a ListBuilder) and only a selection (with no index) has been
+    // provided.
+    if (options[0] === 'Loading...') return
```

Comment:
Should check against same localised string - if that can be changed

Reply:
I'm a bit reluctant to do that as I think it'll break things. It would be okay for the DEFAULT_LOADING_VALUE, but it won't work easily in this case. That's because the "Loading" values are stored in each plugin's `pluginConfig.json` file, which aren't currently localisable. And we can't use the default in this case, because the `options` default loading value is `[ "Loading..." ]` not `"Loading..." (because it needs to be in an options array). 

So I think safer to leave this for now until we come up with a better overall solution for this (but that'll be quite complicated). (Ideally you don't see the "Loading" strings very often in the UI)

Reply:
It might show up in slow connections with bad configured queries (that don't filter the data to show)